packages/runtime/container-runtime/src/garbageCollection.ts

```diff
@@ -787,9 +802,9 @@ export class GarbageCollector implements IGarbageCollector {
         this.updateCurrentState(gcData, gcResult, currentReferenceTimestampMs);
         this.runtime.updateUsedRoutes(gcResult.referencedNodeIds, currentReferenceTimestampMs);
 
-        // Log events for objects that are ready to be deleted by sweep. When we have sweep enabled, we will
-        // delete these objects here instead.
-        this.logSweepEvents(logger, currentReferenceTimestampMs);
+        // If enabled, run Sweep to delete objects that have been unreferenced sufficiently long
+        // This also logs once per node when it is (or  would be) swept
+        this.sweepIfEnabled(logger, currentReferenceTimestampMs);
```

Comment:
Shouldn't this code run only when summary is generated? I understand the desire to go faster, but will it cause trouble? Maybe not, but worth considering all the interesting cases.
It sounds like it's created in every container, including non-summarizing container. This user may not have write permissions to a file. Will it result in some errors? Or all the modifications we do here results in no ops and only reflect back to file through summarizer?

Reply:
It does only run when the summary is generated, if you look at its call site and keep tracing back.

Reply:
But yes, this class is instantiated for all clients. Maybe we should think about if this is necessary and/or expensive.

Reply:
Actually it's kind of nice that it does, because then I can do things like throw an error to close the main container if we try to open a test mode document we don't understand (see top of configForExistingContainer function)